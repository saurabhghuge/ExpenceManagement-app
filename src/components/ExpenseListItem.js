import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt,note }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
<p>{amount}  ---{note}--- {createdAt} </p>
  </div>
);

export default ExpenseListItem;
