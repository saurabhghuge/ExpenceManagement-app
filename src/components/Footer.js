import React from 'react';
import { Link , NavLink} from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Footer = () => (
  <footer className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/dashboard">
          <h1>Expenses Management</h1>
        </Link>
        
      </div>
       <NavLink className = "second__header"  to="/dashboard" activeClassName="is-active" >Dashboard</NavLink>
      <NavLink className = "second__header"  to="/help" activeClassName="is-active">About</NavLink>
      <button className= "button--link" onClick={startLogout}>Logout</button>

    </div>
  </footer>
);