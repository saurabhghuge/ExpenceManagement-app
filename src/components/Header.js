// import React from 'react';
// import { NavLink } from 'react-router-dom';
// import {connect} from 'react-redux';
// import {startLogout} from '../actions/auth';


// export const Header = ({startLogout}) => (
//   <header className = "header"> 
     
//     <div className = "content-container">
//     <h1 className = "header__title">Expenses Management</h1>  
//       <div className = "header__content " >    
//        <div  >
//     {/* <NavLink to="/" activeClassName="is-active" exact={true}>Login Page</NavLink> */}
//     <NavLink className = "second__header"  to="/dashboard" activeClassName="is-active" >Dashboard</NavLink>
//     <NavLink className = "second__header"  to="/help" activeClassName="is-active">About</NavLink>
//     </div>
//     <button onClick ={startLogout}> Logout</button>
//     </div>
//     </div>

//   </header>
// );

// const mapDispatchToProps=(dispatch)=>({
//   startLogout : () => dispatch(startLogout())
// })

// export default connect(undefined, mapDispatchToProps)(Header);


import React from 'react';
import { Link , NavLink} from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/dashboard">
          <h1>Expenses Management</h1>
        </Link>
        
      </div>
       <NavLink className = "second__header"  to="/dashboard" activeClassName="is-active" >Dashboard</NavLink>
      <NavLink className = "second__header"  to="/help" activeClassName="is-active">About</NavLink>
      <button className= "button--link" onClick={startLogout}>Logout</button>

    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
