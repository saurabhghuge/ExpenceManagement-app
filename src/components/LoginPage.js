import React from 'react';
import {connect } from 'react-redux';
import{startLogin} from '../actions/auth';

export const LoginPage = ({startLogin})=>(
     <div className = "box-layout">    
        <div className="box-layout__box">
            <h1 className = "box-layout__title">Expenses Management </h1>
            <p>Add Your Expenses and Manage them without Paper and Calculator </p>
            <button className = "button" onClick = {startLogin}>Login with Google</button>
        </div></div>
);

const mapDispatchToProps = (dispatch) =>({
    startLogin:() => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
