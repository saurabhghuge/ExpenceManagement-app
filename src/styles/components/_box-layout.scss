.box-layout{
    align-items: center;
    background: url('/images/login.jpeg');
    background-size: cover;
    height: 100vh;
    width: 100vw;
    display: flex;
    justify-content: center;
}
 
.box-layout__box{
    background: fade-out($color: #55ccc2, $amount: .15);
    border-radius: 3px;
    padding: $l-size $m-size ;
    text-align: center;
    width: 25rem;

}

.box-layout__title{
    margin: 0 0 $m-size 0;
    line-height: 1;
}