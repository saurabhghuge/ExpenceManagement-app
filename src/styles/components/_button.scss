.button{
    background: rgb(9, 138, 243) ;
    border: none;
    color: white;
    display: inline-block;
    padding: 1rem;
    text-decoration: none;
    font-weight: 300;
    font-size: $m-size;
    margin-bottom: $s-size;
}

.button--link{
    background: none;
    position: absolute;
    text-align: right;
    right: 10%;
    
    
}
.button--remove{
    @extend .button;
    background: rgb(87, 82, 82) ;

}