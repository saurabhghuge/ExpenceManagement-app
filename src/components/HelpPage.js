import React from 'react';
import Header from '../components/Header';
import ContactIcon from '../components/icons/contact';

const HelpPage = () => (
  <div>
    <Header/>
    <div className="main_cont">
    <div className='contact_container'>
      <h1>Contact</h1>
      <h4 className='contacts_text'>Feel free to contact me so we can get your project started!</h4>
      <div className='contacts_subb'>
      <a className='contacts_email' href={'mailto: saurabhghuge1998@gmail.com'}>saurabhghuge1998@gmail.com</a>
      <p>+91-8108381738</p>
      </div>
    <ContactIcon type='contact'/>
    </div>
  </div></div>
);

export default HelpPage;
