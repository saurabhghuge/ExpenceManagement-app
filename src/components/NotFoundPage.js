import React from 'react';
import { Link } from 'react-router-dom';
import Header from '../components/Header';


const NotFoundPage = () => (
    <div>
      <Header/>
      <div className = "page-header">
        <div className = "content-container">
        <h1 className = "page-header__title" ><h1>NOT FOUND</h1> <Link to="/">Go home</Link></h1>
        <div>
        <p>You just hit a route that doesn&#39;t exist... the sadness.</p>
         </div>
        </div>
        </div>
      </div>

);

export default NotFoundPage;
