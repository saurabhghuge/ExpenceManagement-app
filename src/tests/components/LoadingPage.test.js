import React from 'react';
import {shallow} from 'enzyme';
import LoadingPage from '../../components/LoadingPage';
import { LoginPage } from '../../components/LoginPage';

test('shoul correctly render Loading page',() =>{
    const wrapper = shallow(<LoginPage/>);
    expect(wrapper).toMatchSnapshot();
})