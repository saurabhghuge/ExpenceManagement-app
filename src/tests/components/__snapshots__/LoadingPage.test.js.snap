// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoul correctly render Loading page 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expenses Management 
    </h1>
    <p>
      Add Your Expenses and Manage them without Paper and Calculator 
    </p>
    <button
      className="button"
    >
      Login with Google
    </button>
  </div>
</div>
`;
